Neutron NetworkingV2PortSecGroupAssociate ci tests failing due to `invalid memory address or nil pointer dereference`

Scheduled run on Neutron is failing. Example: https://github.com/terraform-provider-openstack/terraform-provider-openstack/actions/runs/4570368393

Failing tests:
- TestAccNetworkingV2PortSecgroupAssociate_import
- TestAccNetworkingV2PortSecGroupAssociate_update
- TestAccNetworkingV2PortSecgroupAssociate_import
- TestAccNetworkingV2PortSecGroupAssociate_update

Logs of failed test:
```
2023-03-31T01:42:50.0843028Z === RUN   TestAccNetworkingV2PortSecGroupAssociate_update
2023-03-31T01:42:50.0847779Z --- FAIL: TestAccNetworkingV2PortSecGroupAssociate_update (0.00s)
2023-03-31T01:42:50.0877452Z panic: runtime error: invalid memory address or nil pointer dereference [recovered]
2023-03-31T01:42:50.0883392Z 	panic: runtime error: invalid memory address or nil pointer dereference
2023-03-31T01:42:50.0888385Z [signal SIGSEGV: segmentation violation code=0x1 addr=0xa8 pc=0xd29618]
2023-03-31T01:42:50.0888602Z 
2023-03-31T01:42:50.0891126Z goroutine 19 [running]:
2023-03-31T01:42:50.0894515Z testing.tRunner.func1.2({0x1145da0, 0x1cc38f0})
2023-03-31T01:42:50.0897301Z 	/opt/hostedtoolcache/go/1.20.2/x64/src/testing/testing.go:1526 +0x24e
2023-03-31T01:42:50.0899472Z testing.tRunner.func1()
2023-03-31T01:42:50.0902324Z 	/opt/hostedtoolcache/go/1.20.2/x64/src/testing/testing.go:1529 +0x39f
2023-03-31T01:42:50.0905802Z panic({0x1145da0, 0x1cc38f0})
2023-03-31T01:42:50.0908538Z 	/opt/hostedtoolcache/go/1.20.2/x64/src/runtime/panic.go:884 +0x213
2023-03-31T01:42:50.0911062Z github.com/gophercloud/utils/terraform/auth.(*Config).LoadAndValidate(0xc00035c6c0)
2023-03-31T01:42:50.0914208Z 	/home/runner/go/pkg/mod/github.com/gophercloud/utils@v0.0.0-20230324070755-05e9e7f5ea4d/terraform/auth/config.go:156 +0x738
2023-03-31T01:42:50.0916799Z github.com/terraform-provider-openstack/terraform-provider-openstack/openstack.testAccAuthFromEnv()
2023-03-31T01:42:50.0919976Z 	/home/runner/work/terraform-provider-openstack/terraform-provider-openstack/openstack/provider_test.go:472 +0x625
2023-03-31T01:42:50.0924896Z github.com/terraform-provider-openstack/terraform-provider-openstack/openstack.testAccCheckNetworkingV2PortSecGroupCreatePort(0xc000419860, {0x12d08c4, 0xb}, 0x1)
2023-03-31T01:42:50.0928649Z 	/home/runner/work/terraform-provider-openstack/terraform-provider-openstack/openstack/resource_openstack_networking_port_secgroup_associate_v2_test.go:134 +0x4e
2023-03-31T01:42:50.0931369Z github.com/terraform-provider-openstack/terraform-provider-openstack/openstack.TestAccNetworkingV2PortSecGroupAssociate_update(0xc000419860)
2023-03-31T01:42:50.0934663Z 	/home/runner/work/terraform-provider-openstack/terraform-provider-openstack/openstack/resource_openstack_networking_port_secgroup_associate_v2_test.go:21 +0x97
2023-03-31T01:42:50.0937942Z testing.tRunner(0xc000419860, 0x1397bc8)
2023-03-31T01:42:50.0940930Z 	/opt/hostedtoolcache/go/1.20.2/x64/src/testing/testing.go:1576 +0x10b
2023-03-31T01:42:50.0942892Z created by testing.(*T).Run
2023-03-31T01:42:50.0945790Z 	/opt/hostedtoolcache/go/1.20.2/x64/src/testing/testing.go:1629 +0x3ea
2023-03-31T01:42:50.0954068Z FAIL	github.com/terraform-provider-openstack/terraform-provider-openstack/openstack	0.019s
```

--- a/openstack/data_source_openstack_networking_trunk_v2.go
+++ b/openstack/data_source_openstack_networking_trunk_v2.go
@@ -119,7 +119,7 @@ func dataSourceNetworkingTrunkV2Read(ctx context.Context, d *schema.ResourceData
 		listOpts.PortID = v.(string)
 	}
 
-	if v, ok := d.GetOkExists("admin_state_up"); ok {
+	if v, ok := d.GetOk("admin_state_up"); ok {
 		asu := v.(bool)
 		listOpts.AdminStateUp = &asu
 	}

--- a/openstack/internal/pathorcontents/read.go
+++ b/openstack/internal/pathorcontents/read.go
@@ -2,7 +2,6 @@
 package pathorcontents
 
 import (
-	"io/ioutil"
 	"os"
 
 	"github.com/mitchellh/go-homedir"
@@ -32,7 +31,7 @@ func Read(poc string) (string, bool, error) {
 	}
 
 	if _, err := os.Stat(path); err == nil {
-		contents, err := ioutil.ReadFile(path)
+		contents, err := os.ReadFile(path)
 		if err != nil {
 			return string(contents), true, err
 		}

--- a/openstack/provider_test.go
+++ b/openstack/provider_test.go
@@ -3,7 +3,6 @@ package openstack
 import (
 	"context"
 	"fmt"
-	"io/ioutil"
 	"os"
 	"testing"
 
@@ -417,7 +416,7 @@ func envVarFile(varName string) (string, error) {
 		return "", err
 	}
 
-	tmpFile, err := ioutil.TempFile("", varName)
+	tmpFile, err := os.CreateTemp("", varName)
 	if err != nil {
 		return "", fmt.Errorf("Error creating temp file: %s", err)
 	}

--- a/openstack/provider_test.go
+++ b/openstack/provider_test.go
@@ -4,12 +4,14 @@ import (
 	"context"
 	"fmt"
 	"os"
+	"strconv"
 	"testing"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 	"github.com/terraform-provider-openstack/terraform-provider-openstack/openstack/internal/pathorcontents"
 
+	"github.com/gophercloud/gophercloud"
 	"github.com/gophercloud/utils/terraform/auth"
 	"github.com/gophercloud/utils/terraform/mutexkv"
 )
@@ -442,29 +444,40 @@ func testAccAuthFromEnv() (*Config, error) {
 		tenantName = os.Getenv("OS_PROJECT_NAME")
 	}
 
+	authOpts := &gophercloud.AuthOptions{
+		Scope: &gophercloud.AuthScope{System: testGetenvBool("OS_SYSTEM_SCOPE")},
+	}
+
 	config := Config{
 		auth.Config{
-			CACertFile:        os.Getenv("OS_CACERT"),
-			ClientCertFile:    os.Getenv("OS_CERT"),
-			ClientKeyFile:     os.Getenv("OS_KEY"),
-			Cloud:             os.Getenv("OS_CLOUD"),
-			DefaultDomain:     os.Getenv("OS_DEFAULT_DOMAIN"),
-			DomainID:          os.Getenv("OS_DOMAIN_ID"),
-			DomainName:        os.Getenv("OS_DOMAIN_NAME"),
-			EndpointType:      os.Getenv("OS_ENDPOINT_TYPE"),
-			IdentityEndpoint:  os.Getenv("OS_AUTH_URL"),
-			Password:          os.Getenv("OS_PASSWORD"),
-			ProjectDomainID:   os.Getenv("OS_PROJECT_DOMAIN_ID"),
-			ProjectDomainName: os.Getenv("OS_PROJECT_DOMAIN_NAME"),
-			Region:            os.Getenv("OS_REGION"),
-			Token:             os.Getenv("OS_TOKEN"),
-			TenantID:          tenantID,
-			TenantName:        tenantName,
-			UserDomainID:      os.Getenv("OS_USER_DOMAIN_ID"),
-			UserDomainName:    os.Getenv("OS_USER_DOMAIN_NAME"),
-			Username:          os.Getenv("OS_USERNAME"),
-			UserID:            os.Getenv("OS_USER_ID"),
-			MutexKV:           mutexkv.NewMutexKV(),
+			CACertFile:                  os.Getenv("OS_CACERT"),
+			ClientCertFile:              os.Getenv("OS_CERT"),
+			ClientKeyFile:               os.Getenv("OS_KEY"),
+			Cloud:                       os.Getenv("OS_CLOUD"),
+			DefaultDomain:               os.Getenv("OS_DEFAULT_DOMAIN"),
+			DomainID:                    os.Getenv("OS_DOMAIN_ID"),
+			DomainName:                  os.Getenv("OS_DOMAIN_NAME"),
+			EndpointType:                os.Getenv("OS_ENDPOINT_TYPE"),
+			IdentityEndpoint:            os.Getenv("OS_AUTH_URL"),
+			Password:                    os.Getenv("OS_PASSWORD"),
+			ProjectDomainID:             os.Getenv("OS_PROJECT_DOMAIN_ID"),
+			ProjectDomainName:           os.Getenv("OS_PROJECT_DOMAIN_NAME"),
+			Region:                      os.Getenv("OS_REGION"),
+			Token:                       os.Getenv("OS_TOKEN"),
+			TenantID:                    tenantID,
+			TenantName:                  tenantName,
+			UserDomainID:                os.Getenv("OS_USER_DOMAIN_ID"),
+			UserDomainName:              os.Getenv("OS_USER_DOMAIN_NAME"),
+			Username:                    os.Getenv("OS_USERNAME"),
+			UserID:                      os.Getenv("OS_USER_ID"),
+			ApplicationCredentialID:     os.Getenv("OS_APPLICATION_CREDENTIAL_ID"),
+			ApplicationCredentialName:   os.Getenv("OS_APPLICATION_CREDENTIAL_NAME"),
+			ApplicationCredentialSecret: os.Getenv("OS_APPLICATION_CREDENTIAL_SECRET"),
+			UseOctavia:                  testGetenvBool("OS_USE_OCTAVIA"),
+			DelayedAuth:                 testGetenvBool("OS_DELAYED_AUTH"),
+			AllowReauth:                 testGetenvBool("OS_ALLOW_REAUTH"),
+			AuthOpts:                    authOpts,
+			MutexKV:                     mutexkv.NewMutexKV(),
 		},
 	}
 
@@ -473,4 +486,9 @@ func testAccAuthFromEnv() (*Config, error) {
 	}
 
 	return &config, nil
+}
+
+func testGetenvBool(env string) bool {
+	ret, _ := strconv.ParseBool(os.Getenv(env))
+	return ret
 }

--- a/openstack/resource_openstack_compute_aggregate_v2.go
+++ b/openstack/resource_openstack_compute_aggregate_v2.go
@@ -84,7 +84,7 @@ func resourceComputeAggregateV2Create(_ context.Context, d *schema.ResourceData,
 		for _, host := range hosts.List() {
 			_, err = aggregates.AddHost(computeClient, aggregate.ID, aggregates.AddHostOpts{Host: host.(string)}).Extract()
 			if err != nil {
-				return diag.Errorf("Error adding host %s to Openstack aggregate: %s", host.(string), err)
+				return diag.Errorf("Error adding host %s to OpenStack aggregate: %s", host.(string), err)
 			}
 		}
 	}
@@ -166,7 +166,7 @@ func resourceComputeAggregateV2Update(_ context.Context, d *schema.ResourceData,
 			log.Printf("[DEBUG] Removing host '%s' from aggregate '%s'", host, d.Get("name"))
 			_, err = aggregates.RemoveHost(computeClient, id, aggregates.RemoveHostOpts{Host: host}).Extract()
 			if err != nil {
-				return diag.Errorf("Error deleting host %s from Openstack aggregate: %s", host, err)
+				return diag.Errorf("Error deleting host %s from OpenStack aggregate: %s", host, err)
 			}
 		}
 		for _, h := range hostsToAdd.List() {
@@ -174,7 +174,7 @@ func resourceComputeAggregateV2Update(_ context.Context, d *schema.ResourceData,
 			log.Printf("[DEBUG] Adding host '%s' to aggregate '%s'", host, d.Get("name"))
 			_, err = aggregates.AddHost(computeClient, id, aggregates.AddHostOpts{Host: host}).Extract()
 			if err != nil {
-				return diag.Errorf("Error adding host %s to Openstack aggregate: %s", host, err)
+				return diag.Errorf("Error adding host %s to OpenStack aggregate: %s", host, err)
 			}
 		}
 	}
@@ -203,20 +203,20 @@ func resourceComputeAggregateV2Delete(_ context.Context, d *schema.ResourceData,
 		return diag.Errorf("Can't convert ID to integer: %s", err)
 	}
 
-	// Openstack do not delete the host aggregate if it's not empty
+	// OpenStack do not delete the host aggregate if it's not empty
 	hostsToDelete := d.Get("hosts").(*schema.Set)
 	for _, h := range hostsToDelete.List() {
 		host := h.(string)
 		log.Printf("[DEBUG] Removing host '%s' from aggregate '%s'", host, d.Get("name"))
 		_, err = aggregates.RemoveHost(computeClient, id, aggregates.RemoveHostOpts{Host: host}).Extract()
 		if err != nil {
-			return diag.Errorf("Error deleting host %s from Openstack aggregate: %s", host, err)
+			return diag.Errorf("Error deleting host %s from OpenStack aggregate: %s", host, err)
 		}
 	}
 
 	err = aggregates.Delete(computeClient, id).ExtractErr()
 	if err != nil {
-		return diag.Errorf("Error deleting Openstack aggregate: %s", err)
+		return diag.Errorf("Error deleting OpenStack aggregate: %s", err)
 	}
 
 	return nil

--- a/openstack/resource_openstack_objectstorage_tempurl_v1_test.go
+++ b/openstack/resource_openstack_objectstorage_tempurl_v1_test.go
@@ -2,7 +2,7 @@ package openstack
 
 import (
 	"fmt"
-	"io/ioutil"
+	"io"
 	"net/http"
 	"testing"
 
@@ -93,7 +93,7 @@ func testAccCheckObjectstorageTempurlV1Get(n string) resource.TestCheckFunc {
 			return fmt.Errorf("Failed to retrieve tempurl: %s", url)
 		}
 		defer resp.Body.Close()
-		data, err := ioutil.ReadAll(resp.Body)
+		data, err := io.ReadAll(resp.Body)
 		if err != nil {
 			return fmt.Errorf("Failed to read tempurl body: %s", url)
 		}

